Remove backward and requires_grad from Autograd backend key

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49613 Remove backward and requires_grad from Autograd backend key**
* #49259 Remove `use_c10_dispatcher: full` lines
* #49251 Remove generated_unboxing_wrappers and setManuallyBoxedKernel
* #49220 Remove .impl_UNBOXED() and functionalities associated with it
* #49164 Remove codegen logic to support non-c10-full ops

Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

smessmer committed:
Remove backward and requires_grad from Autograd backend key

Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

albanD commented on torch/csrc/autograd/VariableTypeManual.cpp:
```diff
@@ -310,14 +310,6 @@ TORCH_LIBRARY_IMPL(aten, Autograd, m) {
-  // kernel, because when called with a VariableTensorId tensor, it goes through the variable fallback kernel,
-  // which calls callBoxed(), which doesn't support optional tensor arguments yet and backward() has an optional
-  // tensor argument.
-  // TODO Once callBoxed() supports optional tensor arguments, we can enable `use_c10_dispatcher: full` for backward()
```
Is c10 dispatcher full enabled in another PR in the stack as per the comment? :D

smessmer replied:
Yes. And, as of today, landed.

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove backward and requires_grad from Autograd backend key"


Just following a TODO in the code base...

Differential Revision: [D25644597](https://our.internmc.facebook.com/intern/diff/D25644597/)

[ghstack-poisoned]
